featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -331,6 +334,72 @@ def is_leap_year(vals):
         return is_leap_year
 
 
+class IsLunchTime(TransformPrimitive):
+    """Determines if a datetime falls during lunch time (non-holiday weekday with hour=12)
+
+    Args:
+        country (str): Country to use for determining Holidays.
+            Default is 'US'. Should be one of the available countries here:
+            https://github.com/dr-prodigy/python-holidays#available-countries
+        include_weekend (bool): Include weekend dates as datetimes with valid
+            lunch times. Default is True.
+        include_holidays (bool): Include holiday dates as datetimes with valid
+            lunch times. Default is False.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2022, 6, 21, 12, 3, 3),
+        ...          datetime(2019, 1, 3, 4, 4, 4),
+        ...          datetime(2022, 1, 1, 12, 1, 2),
+        ...          np.nan]
+        >>> ilt = IsLunchTime()
+        >>> ilt(dates).tolist()
+        [True, False, False, False]
+    """
+
+    name = "is_lunch_time"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+    compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
+    description_template = "whether {} falls during lunch time"
+
+    def __init__(self, country="US", include_weekends=True, include_holidays=False):
```

Comment:
A couple thoughts here...

- I think we should also include the hour as a configurable parameter as I'm not sure the noon hour is always considered lunch time in all contexts. We can default to `hour=12`.
- Just wondering about the logic for having `include_weekends` default to `True`, but `include_holidays` default to False? If we are trying to return only True in situations where it is lunch hour on a "standard" workday, then I think both should default to False.

Reply:
I agree with both of those both points